File: core/src/main/java/cz/dynawest/jtexy/parsers/TexyParserEvent.java

package cz.dynawest.jtexy.parsers;

import cz.dynawest.jtexy.events.TexyEvent;

/**
 *  Parser event; Carries text and parser backref.
 * 
 * @author Ondrej Zizka
 */
public class TexyParserEvent extends TexyEvent {

	/* -- Fields -- */
	
	protected final TexyParser parser;
	public final TexyParser getParser() {		return parser;	}

	

	/** Const */
	public TexyParserEvent(TexyParser parser, String text) {
        super(text);
		this.parser = parser;
	}
	
}

File: core/src/main/java/cz/dynawest/jtexy/parsers/TexyParserEvent.kt
package cz.dynawest.jtexy.parsers

import cz.dynawest.jtexy.events.TexyEvent

/**
 * Parser event; Carries text and parser backref.
 *
 * @author Ondrej Zizka
 */
open class TexyParserEvent
(
              val parser: TexyParser, text: String
) : TexyEvent(text)
